js/ermrest.js

```diff
@@ -62,14 +62,18 @@ var ERMrest = (function (module) {
      * URI should be to the ERMrest _service_. For example,
      * `https://www.example.org/ermrest`.
      */
-    function getServer(uri) {
+    function getServer(uri, params) {
 
         if (uri === undefined || uri === null)
             throw new module.InvalidInputError("URI undefined or null");
 
+        if (typeof params === 'undefined' || params === null) {
+            params = {'cid': null};
```

Comment:
@robes: I asked Jessie to explicitly set it to "null", so we can
distinguish from the log between the version of chaise that forgets to set
cid vs the older version of chaise.


On Fri, Jul 15, 2016 at 4:02 PM, robes <notifications@github.com> wrote:

> In js/ermrest.js
> <https://github.com/informatics-isi-edu/ermrestjs/pull/80#discussion_r71051855>
> :
>
> >
> >          if (uri === undefined || uri === null)
> >              throw new module.InvalidInputError("URI undefined or null");
> >
> > +        if (typeof params === 'undefined' || params === null) {
> > +            params = {'cid': null};
>
> it's okay if params is null. we do not need to initialize it to {'cid':
> null}. just leave it null and downsteam you just won't add the params to
> the url.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/informatics-isi-edu/ermrestjs/pull/80/files/ebbd3935c7fe5ffe5f8ae25416349c3fef2224b5#r71051855>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAZUDHGHungzjCQmV2prLOA4xOu7ooTEks5qWBGggaJpZM4JK4h2>
> .[image: Web Bug from
> https://github.com/notifications/beacon/AAZUDM9uLIw-yoUcdtJI9b0N15_eJdtSks5qWBGggaJpZM4JK4h2.gif]
>